src/Database/Adapter/MariaDB.php

```
@@ -2376,4 +2388,33 @@ public function getSupportForTTLIndexes(): bool
     {
         return false;
     }
+
+    /**
+     * MariaDB/MySQL Upsert: append `ON DUPLICATE KEY UPDATE col = VALUES(col), ...`
+     * so the INSERT replaces the matched row's columns instead of throwing.
+     *
+     * @param array<string> $columns
+     */
+    protected function getInsertSuffix(string $table, array $columns = []): string
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$table'. (undefined)

(UnusedFormalParameter)